### PR TITLE
docs: add types to public methods and getters/setters

### LINKED ISSUE #8832: Missing docs for properties/methods
#### Bug, feature request, or proposal:
Bug/Proposal.

#### What is the expected behavior?
The listed propeties/methods have no docs.

#### What is the current behavior?
Here's the list that I found of all missing docs.

| Class | Properties | Methods |
| ------------- | ------------- | ------------- 
| [**MatAutocompleteTrigger**](https://material.angular.io/components/autocomplete/api#MatAutocompleteTrigger)  |`autocomplete`, `panelOpen`
| [**MatChip**](https://material.angular.io/components/chips/api#MatChip) | `ariaSelected` |
| [**MatChipInput**](https://material.angular.io/components/chips/api#MatChipInput) | `focused` | `focus` 
| [**MatChipList**](https://material.angular.io/components/chips/api#MatChipList) | `controlType` , `ngControl`, `role` | `onContainerClick`, `setDescribedByIds`
| [**MatDialogClose**](https://material.angular.io/components/dialog/api#MatDialogClose) | `dialogRef`
| [**MatDialogRef**](https://material.angular.io/components/dialog/api#MatDialogRef) | `id`
| [**MatDialogTitle**](https://material.angular.io/components/dialog/api#MatDialogTitle) | `id`
| [**MatDatepickerInput**](https://material.angular.io/components/datepicker/api#MatDatepickerInput) | | `registerOnValidatorChange`, `validate`
| [**MatExpansionPanelHeader**](https://material.angular.io/components/expansion/api#MatExpansionPanelHeader) | `panel`
| [**MatError**](https://material.angular.io/components/form-field/api#MatError) | `id`
| [**MatInput**](https://material.angular.io/components/input/api#MatInput) | `empty`, `ngControl`, `shouldLabelFloat` | `focus`, `onContainerClick`, `setDescribedByIds`
| [**MatListOption**](https://material.angular.io/components/list/api#MatListOption) | `selectionList`
| [**MatSelect**](https://material.angular.io/components/select/api#MatSelect)  | `ngControl`, `shouldLabelFloat`
| [**MatSlideToggleChange**](https://material.angular.io/components/slide-toggle/api#MatSlideToggleChange) | All `MatSlideToggleChange` properties |
| [**SimpleSnackBar**](https://material.angular.io/components/snack-bar/api#SimpleSnackBar) | `snackBarRef` 

#### Is there anything else we should know?
Related to #8789.

### PR DESCRIPTION
**Pull request by rafaelss95**
Basically, this PR adds the missing types/docs for leftovers after #7711.

Fixes #8832.

**PS**: It'd be great to use something like [typedef](https://palantir.github.io/tslint/rules/typedef/) to ensure types for getters and setters are specified.

### CODE REVIEW

File: src/lib/chips/chip-list.ts [hunk 1]
```diff
@@ -184,9 +164,7 @@ export class MatChipList extends _MatChipListMixinBase implements MatFormFieldCo
     return this.multiple ? this._selectionModel.selected : this._selectionModel.selected[0];
   }
 
-  get role(): string|null {
-    return this.empty ? null : 'listbox';
-  }
+  get role(): string | null { return this.empty ? null : 'listbox'; }
```
Comment (rafaelss95) [inline src/lib/chips/chip-list.ts]: JSdocs for this?

Comment (jelbourn) [reply src/lib/chips/chip-list.ts]: "The ARIA role applied to the chip list"

**Comment by rafaelss95** [comment]
@jelbourn Since there are only changes in docs, wouldn't it be better if you could review and merge it ASAP to prevent the repeated rebase? Do you guys have a certain day to review this kind of PR?

**Comment by jelbourn** [comment]
The problem is that anything that touches code (including the types) has to go through Google's presubmit process, which means it gets batched in with other PRs. Anything that gets merged earlier in the pipeline would have a potential conflict.

**Comment by rafaelss95** [comment]
Got it. How do you suggest to break them? What's the criteria?

**Pull request closed without merging**